geomesa-compute/src/main/scala/org/locationtech/geomesa/compute/spark/GeoMesaSpark.scala

```diff
@@ -22,25 +22,67 @@ import java.util.UUID
 import com.esotericsoftware.kryo.Kryo
 import com.esotericsoftware.kryo.io.{Input, Output}
 import com.google.common.cache.{CacheBuilder, CacheLoader}
-import org.apache.accumulo.core.client.mapreduce.AccumuloInputFormat
+import com.typesafe.scalalogging.slf4j.Logging
 import org.apache.accumulo.core.client.mapreduce.lib.util.{ConfiguratorBase, InputConfigurator}
+import org.apache.accumulo.core.client.mapreduce.{AccumuloInputFormat, RangeInputSplit}
 import org.apache.accumulo.core.data.{Key, Value}
 import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.io.Text
+import org.apache.hadoop.mapreduce.{InputSplit, Job, JobContext}
 import org.apache.spark.rdd.RDD
 import org.apache.spark.serializer.KryoRegistrator
 import org.apache.spark.{SparkConf, SparkContext}
 import org.geotools.data.{DataStore, DataStoreFinder, DefaultTransaction, Query}
 import org.geotools.factory.CommonFactoryFinder
+import org.geotools.filter.text.ecql.ECQL
 import org.locationtech.geomesa.core.data._
-import org.locationtech.geomesa.core.index.{ExplainPrintln, STIdxStrategy, _}
+import org.locationtech.geomesa.core.index._
 import org.locationtech.geomesa.feature._
 import org.locationtech.geomesa.feature.kryo.{KryoFeatureSerializer, SimpleFeatureSerializer}
+import org.locationtech.geomesa.jobs.GeoMesaConfigurator
+import org.locationtech.geomesa.jobs.mapreduce.{GeoMesaInputFormat, GroupedSplit}
 import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
 import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
 
 import scala.collection.JavaConversions._
 
-object GeoMesaSpark {
+object GeoMesaSpark extends Logging {
+
+  class GeoSparkInputFormat extends GeoMesaInputFormat
+  {
+    private def init(conf: Configuration): Unit = if (sft == null) {
+      val params = GeoMesaConfigurator.getDataStoreInParams(conf)
+      val ds = DataStoreFinder.getDataStore(params).asInstanceOf[AccumuloDataStore]
+      sft = ds.getSchema(GeoMesaConfigurator.getFeatureType(conf))
+      encoding = ds.getFeatureEncoding(sft)
+      numShards = IndexSchema.maxShard(ds.getIndexSchemaFmt(sft.getTypeName))
+     }
+
+    override def getSplits(context: JobContext) : java.util.List[InputSplit] = {
+      init(context.getConfiguration)
+      val desiredSplits = context.getConfiguration.getInt("SplitsDesired", 10*numShards)
+      val accumuloSplits = delegate.getSplits(context)
+      // try to create 2 mappers per node - account for case where there are less splits than shards
+      val groupSize = Math.max(1, accumuloSplits.length / desiredSplits)
+      logger.error("I am being used....")
```

Comment:
Oops, left my debug in. It wasn't clear whether mapred or mapreduce was being used, so I was worried that there might be name resolution issues.